apps/vc-api/test/vc-api/exchanges/consent-and-resident-card-credentialexchange/consent-and-resident-card.e2e-suite.ts

```diff
@@ -0,0 +1,320 @@
+      vp: returnVp
+    };
+    await walletClient.addSubmissionReview(
+      exchange.getExchangeId(),
+      transactionId,
+      submissionReviewResidentCard
+    );
+    // const submissionReview: SubmissionReviewDto = {
+    //   result: ReviewResult.approved,
+    //   vp: issuedVPConsentCredential
+    // };
+    // await walletClient.addSubmissionReview(exchange.getExchangeId(), transactionId, submissionReview);
+
+    // As the holder, check for a reviewed submission
+    const secondContinuationResponse = await walletClient.continueExchange(
+      issuanceExchangeContinuationEndpoint,
+      didAuthVp,
+      false
+    );
+    const issuedVc1 = secondContinuationResponse.vp.verifiableCredential[0];
+    const issuedVc2 = secondContinuationResponse.vp.verifiableCredential[1];
+    expect(issuedVc1).toBeDefined();
+    expect(issuedVc2).toBeDefined();
+
+    // Configure presentation exchange
+    // POST /exchanges
+    const presentationExchange = new ConsentAndResidentCardPresentation(callbackUrl);
+    const presentationCallbackScope = nock(callbackUrlBase).post(callbackUrlPath).reply(201);
+    const exchangeDef = presentationExchange.getExchangeDefinitionV1();
+    await request(app.getHttpServer()).post(`${vcApiBaseUrl}/exchanges`).send(exchangeDef).expect(201);
+
+    // Start presentation exchange
+    // POST /exchanges/{exchangeId}
+    const exchangeEndpoint = `${vcApiBaseUrl}/exchanges/${presentationExchange.getExchangeIdV1()}`;
+    const presentationVpRequest = await walletClient.startExchange(
+      exchangeEndpoint,
+      presentationExchange.queryType
+    );
+    const presentationExchangeContinuationEndpoint = getContinuationEndpoint(presentationVpRequest);
+    expect(presentationExchangeContinuationEndpoint).toContain(exchangeEndpoint);
+
+    // Holder should parse VP Request for correct credentials...
+    // Assume that holder figures out which VC they need and can prep presentation
+    const presentation: PresentationDto = {
+      '@context': [
+        'https://www.w3.org/2018/credentials/v1',
+        'https://www.w3.org/2018/credentials/examples/v1'
+      ],
+      type: ['VerifiablePresentation'],
+      verifiableCredential: [issuedVc1, issuedVc2],
+      holder: holderDIDDoc.id
+    };
+    const presentationOptions: ProvePresentationOptionsDto = {
+      verificationMethod: holderVerificationMethod,
+      proofPurpose: ProofPurpose.authentication,
+      created: '2021-11-16T14:52:19.514Z',
+      challenge: presentationVpRequest.challenge
+    };
+    const vp = await walletClient.provePresentation({ presentation, options: presentationOptions });
+
+    // Holder submits presentation
+    await walletClient.continueExchange(presentationExchangeContinuationEndpoint, vp, false);
+    presentationCallbackScope.done();
+  });
+
+  it('should be able to verify VP with Consent and ResidentCard credential, where PD has one submission_requirement and input_desciptor group', async () => {
```

Comment:
There seems to be quite a bit of overlap between this test case and the one above. How hard would it be to extract the common portion to reduce the length of the code? I think it is just the exchange definitions that should be changing, right?